### ISSUE TITLE
Focus indicator must stay in the modal until dismissed

### ISSUE THREAD

**Issue by dshokouhi** [home-assistant/frontend#4695]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

The focus indicator must stay within the open modal as long as it is open.  This is important because keyboard only users can easily lose their place and navigate outside the more info panel with no way to return to it.

## Expected behavior

The focus indicator should stay within the modal until dismissed.  When the modal is dismissed the user should then be brought back to the last item that was focused in the previous screen (i.e. the click that brought them there).

## Steps to reproduce

1. Login to HA
2. Open up more info for any panel
3. Press tab key until you leave the modal
4. Attempt to navigate back to the modal via tab and shift+tab

## Environment

- Home Assistant release with the issue: 0.105.0dev
- Last working Home Assistant release (if known): N/A
- Frontend version: 20200130.0 - latest
- UI Type (States or Lovelace): Lovelace
- Browser and browser version: Chrome 79
- Operating system: Windows 10

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

## Additional information

Video: https://drive.google.com/open?id=1ASrN14OlbRHX3GbEHMM6v52jee0eO-i-

A good example of the functionality at play can be found here: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

The delivery address modal is a good example as it prevents keyboard users from losing where they are.

Also the cog wheel on this modal opens a different modal that has the same exact issue:
![image](https://user-images.githubusercontent.com/1634145/73505444-61e0f480-4387-11ea-88bd-b88fa6fefc6a.png)

**Comment by zsarnett** [comment]
This is supported by the new dialogs. Older paper dialogs should be converted and is a separate issue
